From ac9a99505fd768b66dd92e9091e80b576cabc00d Mon Sep 17 00:00:00 2001
From: Sangchul Lee <sangchul1011@gmail.com>
Date: Sat, 27 Aug 2016 21:33:20 +0900
Subject: [PATCH 6/6] sink-input: Remove pa_sink_input_set_volume_ramp()

Please use pa_sink_input_add_volume_ramp_factor() or
pa_sink_input_remove_volume_ramp_factor() instead of it.

Signed-off-by: Sangchul Lee <sc11.lee@samsung.com>
---
 src/pulsecore/sink-input.c | 23 -----------------------
 src/pulsecore/sink-input.h |  1 -
 2 files changed, 24 deletions(-)

diff --git a/src/pulsecore/sink-input.c b/src/pulsecore/sink-input.c
index 6f89aa1..4c7c568 100644
--- a/src/pulsecore/sink-input.c
+++ b/src/pulsecore/sink-input.c
@@ -1475,29 +1475,6 @@ int pa_sink_input_remove_volume_ramp_factor(pa_sink_input *i, const char *key, b
     return 0;
 }
 
-/* Called from main thread */
-void pa_sink_input_set_volume_ramp(
-        pa_sink_input *i,
-        const pa_cvolume_ramp *ramp,
-        bool send_msg) {
-    pa_sink_input_assert_ref(i);
-    pa_assert_ctl_context();
-    pa_assert(PA_SINK_INPUT_IS_LINKED(i->state));
-    pa_assert(ramp);
-
-    pa_cvolume_ramp_convert(ramp, &i->ramp, i->sample_spec.rate);
-
-    pa_log_debug("setting volume ramp with target vol:%d and length:%ld",
-		 i->ramp.ramps[0].target,
-		 i->ramp.ramps[0].length);
-
-
-    /* This tells the sink that volume ramp changed */
-    if (send_msg)
-        pa_assert_se(pa_asyncmsgq_send(i->sink->asyncmsgq, PA_MSGOBJECT(i), PA_SINK_INPUT_MESSAGE_SET_VOLUME_RAMP,
-                                       NULL, 0, NULL) == 0);
-}
-
 /* Called from main context */
 static void set_real_ratio(pa_sink_input *i, const pa_cvolume *v) {
     pa_sink_input_assert_ref(i);
diff --git a/src/pulsecore/sink-input.h b/src/pulsecore/sink-input.h
index 5430d53..5fef3d5 100644
--- a/src/pulsecore/sink-input.h
+++ b/src/pulsecore/sink-input.h
@@ -381,7 +381,6 @@ void pa_sink_input_set_volume(pa_sink_input *i, const pa_cvolume *volume, bool s
 void pa_sink_input_add_volume_factor(pa_sink_input *i, const char *key, const pa_cvolume *volume_factor);
 int pa_sink_input_remove_volume_factor(pa_sink_input *i, const char *key);
 pa_cvolume *pa_sink_input_get_volume(pa_sink_input *i, pa_cvolume *volume, bool absolute);
-void pa_sink_input_set_volume_ramp(pa_sink_input *i, const pa_cvolume_ramp *ramp, bool send_msg);
 void pa_sink_input_add_volume_ramp_factor(pa_sink_input *i, const char *key, const pa_cvolume_ramp *ramp_factor, bool send_msg);
 int pa_sink_input_remove_volume_ramp_factor(pa_sink_input *i, const char *key, bool send_msg);
 
-- 
1.9.1

